Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 75328388f8a6233578ca751d04e3716dc5450d52
Number of Error(s): 4
/mnt/disk1/home/vamosax/smoketest/PR-12350/HPCC-Platform/ecl/hthor/hthor.cpp:8073:34: error: \xe2\x80\x98strisame\xe2\x80\x99 was not declared in this scope
make[2]: *** [ecl/hthor/CMakeFiles/hthor.dir/hthor.cpp.o] Error 1
make[1]: *** [ecl/hthor/CMakeFiles/hthor.dir/all] Error 2
make: *** [all] Error 2

[Build error:](https://gist.github.com/9a965d51c085e6ca5f7eca20ac97528f)
Build log [Build log](https://gist.github.com/HPCCSmoketest/9a965d51c085e6ca5f7eca20ac97528f#file-result-19-03-25-16-14-20-log)

--- a/dali/base/dadfs.hpp
+++ b/dali/base/dadfs.hpp
@@ -814,6 +814,8 @@ inline bool isPartTLK(IPropertyTree &pt) { const char *kind = pt.queryProp("@kin
 inline bool isPartTLK(IDistributedFilePart *p) { return isPartTLK(p->queryAttributes()); }
 inline bool isPartTLK(IPartDescriptor *p) { return isPartTLK(p->queryProperties()); }
 
+inline const char *queryFileKind(IDistributedFile *f) { return queryFileKind(f->queryAttributes()); }
+
 extern da_decl void ensureFileScope(const CDfsLogicalFileName &dlfn, unsigned timeoutms=INFINITE);
 
 extern da_decl bool checkLogicalName(const char *lfn,IUserDescriptor *user,bool readreq,bool createreq,bool allowquery,const char *specialnotallowedmsg);

--- a/dali/base/dafdesc.hpp
+++ b/dali/base/dafdesc.hpp
@@ -23,6 +23,7 @@
 #endif
 
 #include "jiface.hpp"
+#include "jptree.hpp"
 #include "mpbase.hpp"
 class RemoteFilename;
 class RemoteFilenameArray;
@@ -371,5 +372,7 @@ inline DFD_OS SepCharBaseOs(char c)
 
 extern da_decl void extractFilePartInfo(IPropertyTree &info, IFileDescriptor &file);
 
+inline const char *queryFileKind(IPropertyTree &pt) { return pt.queryProp("@kind"); }
+inline const char *queryFileKind(IFileDescriptor *f) { return queryFileKind(f->queryProperties()); }
 
 #endif

--- a/ecl/hthor/hthor.cpp
+++ b/ecl/hthor/hthor.cpp
@@ -8064,16 +8064,21 @@ void CHThorDiskReadBaseActivity::ready()
     if (ldFile)
         dFile = ldFile->queryDistributedFile();  // Null for local file usage
 
-    Linked<IOutputMetaData> publishedMeta = actualDiskMeta;
-    unsigned publishedCrc = actualCrc;
+    Owned<IOutputMetaData> publishedMeta;
+    unsigned publishedCrc = 0;
+    RecordTranslationMode translationMode = getLayoutTranslationMode();
     if (dFile)
     {
-        IPropertyTree &props = dFile->queryAttributes();
-        publishedMeta.setown(getDaliLayoutInfo(props));
-        if (publishedMeta)
-            publishedCrc = props.getPropInt("@formatCrc");
+        const char *kind = queryFileKind(dFile);
+        if (strisame(kind, "flat") || (RecordTranslationMode::AlwaysDisk == translationMode))
+        {
+            IPropertyTree &props = dFile->queryAttributes();
+            publishedMeta.setown(getDaliLayoutInfo(props));
+            if (publishedMeta)
+                publishedCrc = props.getPropInt("@formatCrc");
+        }
     }
-    translators.setown(::getTranslators("hthor-diskread", expectedCrc, expectedDiskMeta, publishedCrc, publishedMeta, projectedCrc, projectedDiskMeta, getLayoutTranslationMode()));
+    translators.setown(::getTranslators("hthor-diskread", expectedCrc, expectedDiskMeta, publishedCrc, publishedMeta, projectedCrc, projectedDiskMeta, translationMode));
     if (translators)
     {
         translator = &translators->queryTranslator();

--- a/system/jlib/jstring.hpp
+++ b/system/jlib/jstring.hpp
@@ -574,6 +574,7 @@ extern jlib_decl bool endsWithIgnoreCase(const char* src, const char* dst);
 inline bool strieq(const char* s, const char* t) { return stricmp(s,t)==0; }
 inline bool streq(const char* s, const char* t) { return strcmp(s,t)==0; }
 inline bool strsame(const char* s, const char* t) { return (s == t) || (s && t && strcmp(s,t)==0); }  // also allow nulls
+inline bool strisame(const char* s, const char* t) { return (s == t) || (s && t && stricmp(s,t)==0); }  // also allow nulls
 inline bool isEmptyString(const char *text) { return !text||!*text; }
 inline bool hasPrefix(const char * text, const char * prefix, bool caseSensitive)
 {

--- a/thorlcr/activities/diskread/thdiskreadslave.cpp
+++ b/thorlcr/activities/diskread/thdiskreadslave.cpp
@@ -68,14 +68,25 @@ protected:
     // return a ITranslator based on published format in part and expected/format
     ITranslator *getTranslators(IPartDescriptor &partDesc)
     {
-        unsigned projectedFormatCrc = helper->getProjectedFormatCrc();
-        IOutputMetaData *projectedFormat = helper->queryProjectedDiskRecordSize();
-        IPropertyTree const &props = partDesc.queryOwner().queryProperties();
-        Owned<IOutputMetaData> publishedFormat = getDaliLayoutInfo(props);
-        unsigned publishedFormatCrc = (unsigned)props.getPropInt("@formatCrc", 0);
         RecordTranslationMode translationMode = getTranslationMode(*this);
+
         unsigned expectedFormatCrc = helper->getDiskFormatCrc();
         IOutputMetaData *expectedFormat = helper->queryDiskRecordSize();
+
+        Linked<IOutputMetaData> publishedFormat;
+        unsigned publishedFormatCrc = 0;
+
+        const char *kind = queryFileKind(&partDesc.queryOwner());
+        if (strisame(kind, "flat") || (RecordTranslationMode::AlwaysDisk == translationMode))
+        {
+            IPropertyTree const &props = partDesc.queryOwner().queryProperties();
+            publishedFormat.setown(getDaliLayoutInfo(props));
+            if (publishedFormat)
+                publishedFormatCrc = (unsigned)props.getPropInt("@formatCrc", 0);
+        }
+
+        unsigned projectedFormatCrc = helper->getProjectedFormatCrc();
+        IOutputMetaData *projectedFormat = helper->queryProjectedDiskRecordSize();
         return ::getTranslators("rowstream", expectedFormatCrc, expectedFormat, publishedFormatCrc, publishedFormat, projectedFormatCrc, projectedFormat, translationMode);
     }
 public:

